app/controllers/income/estimatedPay/update/IncomeUpdatePayslipAmountController.scala

```diff
@@ -112,45 +122,55 @@ class IncomeUpdatePayslipAmountController @Inject()(
     val mandatoryKeys = Seq(UpdateIncome_IdKey, UpdateIncome_NameKey)
     val optionalKeys = Seq(UpdateIncome_PayPeriodKey, UpdateIncome_OtherInDaysKey, UpdateIncome_TaxablePayKey)
 
-    journeyCacheService.collectedValues(mandatoryKeys, optionalKeys) map
-      tupled { (mandatorySeq, optionalSeq) =>
+    journeyCacheService.collectedJourneyValues(mandatoryKeys, optionalKeys) map
+      tupled { (mandatorySeqEither, optionalSeq) =>
         {
-          val viewModel = {
-            val employer = IncomeSource(id = mandatorySeq(0).toInt, name = mandatorySeq(1))
-            val payPeriod = optionalSeq(0)
-            val payPeriodInDays = optionalSeq(1)
-            val taxablePayKey = optionalSeq(2)
-
-            val form = TaxablePayslipForm.createForm().fill(TaxablePayslipForm(taxablePayKey))
-            TaxablePaySlipAmountViewModel(form, payPeriod, payPeriodInDays, employer)
+          mandatorySeqEither match {
+            case Right(mandotorySeq) =>
+              val viewModel = {
+                val incomeSource = IncomeSource(id = mandotorySeq(0).toInt, name = mandotorySeq(1))
+                val payPeriod = optionalSeq(0)
+                val payPeriodInDays = optionalSeq(1)
+                val taxablePayKey = optionalSeq(2)
+
+                val form = TaxablePayslipForm.createForm().fill(TaxablePayslipForm(taxablePayKey))
+                TaxablePaySlipAmountViewModel(form, payPeriod, payPeriodInDays, incomeSource)
+              }
+              Ok(views.html.incomes.taxablePayslipAmount(viewModel))
+
+            case Left(_) => Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())
+
           }
-          Ok(views.html.incomes.taxablePayslipAmount(viewModel))
         }
       }
   }
 
   def handleTaxablePayslipAmount: Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
     implicit val user = request.taiUser
 
-    val employerFuture = IncomeSource.create(journeyCacheService)
+    val incomeSourceFuture = IncomeSource.create(journeyCacheService)
 
     val futurePayPeriod = journeyCacheService.currentValue(UpdateIncome_PayPeriodKey)
     val futurePayPeriodInDays = journeyCacheService.currentValue(UpdateIncome_OtherInDaysKey)
     val futureTotalSalary = journeyCacheService.currentValue(UpdateIncome_TotalSalaryKey)
```

Comment:
So there's a little bug here, when you declare a Future it immediately starts. If employerFuture returns a Left, the values futurePayPeriod, futurePayPeriodInDays and futureTotalSalary will return an error. 

Inlining them all should fix this as you want employerFuture to resolve before proceeding.

Reply:
done